### PR TITLE
Document point.withinBBox

### PR DESCRIPTION
**Pull request by pontusmelke**
Instead of using inequality operators for spatial points queries should now be using the new `point.withinBBox` function.

Builds on top of https://github.com/neo4j/neo4j-documentation/pull/1266

### CODE REVIEW

**Comment by pontusmelke** [comment]
[Add new method for bounding box contains for points](https://trello.com/c/Y7hltkd6/3064-add-new-method-for-bounding-box-contains-for-points)

File: cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc [hunk 1]
```diff
@@ -148,6 +148,35 @@ OPTIONS "{" btree-option: btree-value[, ...] "}"
 a|
 Node key and uniqueness constraints with b-tree options are deprecated and need to be migrated before upgrading to 5.0.
 
+a|
+label:functionality[]
+label:deprecated[]
+[[source, cypher, role="noheader"]
```
Comment (davidoliverSP2) [inline cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc]:
```suggestion
[source, cypher, role="noheader"]
```

File: cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc [hunk 2]
```diff
@@ -148,6 +148,35 @@ OPTIONS "{" btree-option: btree-value[, ...] "}"
...
+ Replaced by:
+ [source, cypher, role="noheader"]
+ ----
+ point.distance(n.prop, point({x:0, y:0})
+ ----
+
+ a|
+ label:functionality[]
+ label:deprecated[]
+ [[source, cypher, role="noheader"]
```
Comment (davidoliverSP2) [inline cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc]:
```suggestion
 [source, cypher, role="noheader"]
```

**Review by davidoliverSP2 — CHANGES_REQUESTED** [review]
A couple of typos to resolve, and this should build docs cleanly. Will try to build again once I've commit these fixes.

**Comment by davidoliverSP2** [comment]
@pontusmelke I've just merged #1266 which I think means this PR can be rebased (I suspect the merge conflicts relate to the commits I pushed on that PR, so rebasing here should also resolve those - apologies if I've made this overly complicated though!)
I believe you had some questions about formatting these updates, so we can have a chat once the PR is building cleanly again.

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialTest.scala [hunk 3]
```diff
@@ -288,68 +288,11 @@ class SpatialTest extends DocumentingTest {
     section("Comparability and Orderability", "cypher-comparability-orderability") {
       p(
         """
-          |Points with different CRS are not comparable.
-          |This means that any function operating on two points of different types will return `null`.
-          |This is true of the <<functions-distance, distance function>> as well as inequality comparisons.
-          |If these are used in a predicate, they will cause the associated `MATCH` to return no results.
+          |The orderability and comparability is due to change in the upcoming 5.0 release.
+          |This means that queries that rely on the comparing two points using the inequality operators, `<`, `<=`, `>`, and `>=`, or the specific order of an `ORDER BY n.point` query needs to be rewritten.
+          |The easiest way is to specify the ordering explicitly using for example `point.x`, `point.y` in _cartesian coordinates_ or `point.longitude` and point.latitude` in `geographic coordinates`.
```
Comment (davidoliverSP2) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialTest.scala]: We've been working towards moving text-only paragraphs out of scala tests, as it makes it a little too difficult for us to format. Until then though, I would suggest rewording this section to:

----
Comparability and orderability 

The comparability and orderability of spacial values are due to change in an upcoming future release. This means that queries that rely on the comparison of two points using the inequality operators, `<`, `⇐`, `>`, and `>=`, or the specific order of an `ORDER BY n.point` query will need to be rewritten.

The most efficient way to do this is to explicitly specify the ordering. For example, by using `point.x`, `point.y` in _cartesian coordinates_, or `point.longitude` and `point.latitude` in _geographic coordinates_.

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialFunctionsTest.scala [hunk 4]
```diff
@@ -130,6 +131,52 @@ class SpatialFunctionsTest extends DocumentingTest {
         resultTable()
       }
     }
+    section("point.withinBBox()", "functions-withinBBox") {
+      p(
+        """`point.withinBBox()` takes three arguments, the first argument is the point to check and the two other define the lower-left (or south-west) and upper-right (or north-east) point of a bounding box respectively.
+          | The return value will be `true` if the provided point is contained in the bounding box (boundary included) otherwise `false`.
```
Comment (davidoliverSP2) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialFunctionsTest.scala]: Suggested rewording:

----

`point.withinBBox()` takes the following arguments: 

* The point to check.
* The lower-left (south-west) point of a bounding box.
* The upper-right (or north-east) point of a bounding box. 
 
The return value will be true if the provided point is contained in the bounding box (boundary included), otherwise the return value will be false.

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialFunctionsTest.scala [hunk 5]
```diff
@@ -130,6 +131,52 @@ class SpatialFunctionsTest extends DocumentingTest {
...
+         "`point.withinBBox` will handle crossing the 180th meridian in geographic coordinates.",
+         "Switching the longitude of the `lowerLeft` and `upperRight` in geographic coordinates will switch the direction of the resulting bounding box.",
+         "Switching the latitude of the `lowerLeft` and `upperRight` in geographic coordinates so that the former is north of the latter will result in an empty range.",
+      )
+      query(
+        """WITH point({x: 0, y: 0, crs: 'cartesian'}) AS lowerLeft, point({x: 10, y: 10, crs: 'cartesian'}) AS upperRight
+          |RETURN point.withinBBox(point({x: 5, y: 5, crs: 'cartesian'}), lowerLeft, upperRight) AS result""".stripMargin, ResultAssertions((r) => {
+          r.toList.head("result").asInstanceOf[Boolean] shouldBe true
+        })) {
+        p("Checking if a point in _Cartesian_ CRS is contained in the bounding box.")
```
Comment (davidoliverSP2) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialFunctionsTest.scala:157]: I think this paragraph can go above the code example (same with the subsequent examples)

**Review by davidoliverSP2 — CHANGES_REQUESTED** [review]
Thanks @pontusmelke 

There was an issue with how the table in the _Deprecations, additions and compatibility_ page handled labels, and I've pushed a fix.

I had a few more suggested editorial changes, but these landed in the scala tests. If your hands full next week, I am happy to try and edit the scala tests involved - please do let me know!

File: cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc [hunk 6]
```diff
@@ -177,12 +177,27 @@ label:deprecated[]
...
+a|
+label:functionality[]
+label:deprecated[]
+[source, cypher, role="noheader"]
+----
+point({x:0, y:0}) <= point({x:1, y:1}) <= point({x:2, y:2})
+----
+a|
+Using inequality operators `<`, `<=`, `>`, and `>=` on spatial points is deprecated.
+Please instea use:
```
Comment (davidoliverSP2) [inline cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc]:
```suggestion
Please instead use:
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialFunctionsTest.scala (hunk 4 above)
Comment (pontusmelke) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialFunctionsTest.scala]: 👍  please verify that I got formatting right?

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialFunctionsTest.scala (hunk 5 above)
Comment (pontusmelke) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialFunctionsTest.scala:157]: All the other examples in this file have `p("...")` just before the `resultTable()`?

Comment (davidoliverSP2) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialFunctionsTest.scala:157]: ah yep - best to be consistent 👍

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialFunctionsTest.scala (hunk 4 above)
Comment (davidoliverSP2) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialFunctionsTest.scala]: Thanks! one small edit to fix the line below the bullets.

**Review by davidoliverSP2 — APPROVED** [review]
